rl4co/envs/routing/tsp.py

```diff
@@ -161,6 +174,50 @@ def check_solution_validity(td: TensorDict, actions: torch.Tensor):
             == actions.data.sort(1)[0]
         ).all(), "Invalid tour"
 
+    @staticmethod
+    def improve_solution(td: TensorDict, actions: torch.Tensor, **kwargs) -> torch.Tensor:
```

Comment:
(as per above, we can call this `local_search`)

---

CC: @leonlan this is what we mentioned during our meeting about having an optional local search operator API in RL4CO with `pyvrp` ;  any thoughts about this?

Reply:
We are unfortunately dropping support for the TSP-based `TwoOpt` in the next release (v0.8.0), because this operator is not very effective in VRPs. I spoke to @fedebotu in Slack and these are my suggestions:
- If you want a dedicated TwoOpt operator for TSP, you can implement this yourself. This should not be too hard (GPT-4 can give the right solution) and you can make this very performant when implemented in Cython. 
- If you want to have local search support for VRPs, the current setup can serve as a good basis. However, some details (e.g., which penalties to set in the CostEvaluator?) need to be addressed. You can always ping me to have a look when you have an implementation.

For now, I assume that you continue to plan using PyVRP v0.7.0 which has support for TwoOpt.

Reply:
About PyVRP (CVRP, CVRPTW, new variants): we can help with that!
About PDP: from what I know, one-on-one pickup and delivery problems are not available right now, is that correct @leonlan ?

Reply:
One-to-one PDP is not supported indeed.

Reply:
Do you have any plan for supporting one-to-one pdp by any chance?
If so, when do you think will it be?

Reply:
There are no plans for supporting PDP in the near future.